scripts/ImportExportTest.sh

```diff
@@ -224,29 +231,35 @@ do
     set +e
     OUTPUT=$("$SPLITSOURCES" "$solfile")
     SPLITSOURCES_RC=$?
-    set -euo pipefail
+    set -e
     if [ ${SPLITSOURCES_RC} == 0 ]
     then
         NSOURCES=$((NSOURCES - 1))
-        for i in $OUTPUT;
+        # shellcheck disable=SC2034
+        for i in $OUTPUT
         do
-            testImportExportEquivalence "$i" "$OUTPUT"
             NSOURCES=$((NSOURCES + 1))
         done
+        testImportExportEquivalence "$solfile" "${OUTPUT[@]}"
```

Comment:
True, running it for each file seems redundant. I think that the idea here was that you compile each file only with those files it actually depends on. There's no mechanism for checking that though and the script just passes in all the files :)

For some context, originally this had a bug where `$OUTPUT` was not quoted so it was only actually using the first file from the list (f712662a663e98733f9ad33ed141a25558921db7). It could be that the author noticed that it's not compiling all files but could not figure out why and made it a loop for a good measure. Or maybe he implemented it with just `$i` at first, then noticed that files are also needed but did not realize that it makes running in a loop pointless :)

In any case, your change here seems fine. I'd put it in a separate commit though instead of missing it with the other stuff. Overall I think that keeping small commits, one per logical change is better for reviewing.

Reply:
Ah ok! Interesting! 

> Overall I think that keeping small commits, one per logical change is better for reviewing.

Yeah that's true!